### PR TITLE
rek expand testdatagen

### PR DESCRIPTION
**Pull request by kilbergr**
## Description

This PR adds options to how the generated test data will look.  To determine the data scenario you'd like to use, check out scenarios in the `testdatagen` package. Each scenario contains a description of what data will be created when the scenario is run. Pass the scenario in as a flag to the generate-test-data function. A sample command: `./bin/generate-test-data -scenarioTwo=true`.
## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Verification Steps

* [ ] All tests pass.
* [ ] Run a number of scenarios and check your db.

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/155344379) for this change

### CODE REVIEW

File: pkg/testdatagen/make_tsp_performance_records.go [hunk 1]
```diff
@@ -49,17 +49,31 @@ func MakeTSPPerformanceData(db *pop.Connection) {
...
+		var awards int
 		minBvs := (qualityBand - 1) * 25
-		bvs := rand.Intn(25) + minBvs
+		bvs := 100 - (rand.Intn(25) + minBvs)
+		// Set rounds according to the flag passed in
+		if rounds == "none" {
+			awards = 0
+		} else if rounds == "full" {
+			awards = models.AwardsPerQualityBand[qualityBand]
+		} else {
```
Comment (breanneboland) [inline pkg/testdatagen/make_tsp_performance_records.go]: Is this else meant to cover the "half" flag value?

**Comment by breanneboland** [comment]
I copied your example command and got this: 
```
$ ./bin/generate-test-data -rounds=half -numTSP=6
flag provided but not defined: -rounds
Usage of ./bin/generate-test-data:
  -config-dir="config": The location of server config files
  -env="development": The environment to run in, configures the database, presently.
```
It does the same for numTSP if I use only that flag. Prereqs and packages are up to date (y'know, just in case). Am I missing something?

File: cmd/generate_test_data/main.go [hunk 2]
```diff
@@ -24,9 +26,9 @@ func main() {
 
 	// Can this be less repetitive without being overly clever?
 	testdatagen.MakeTDLData(db)
-	testdatagen.MakeTSPData(db)
+	testdatagen.MakeMoreTSP(db, *numTSP)
```
Comment (akostibas) [inline cmd/generate_test_data/main.go]: `MakeTSPs` seems more reasonable

File: pkg/models/transportation_service_provider_performance.go [hunk 3]
```diff
@@ -11,6 +11,14 @@ import (
...
 )
 
+// AwardsPerQualityBand is a map of the number of shipments to be awarded per round to each quality band
+var AwardsPerQualityBand = map[int]int{
+	1: 5,
+	2: 3,
+	3: 2,
+	4: 1,
+}
+
```
Comment (akostibas) [inline pkg/models/transportation_service_provider_performance.go]: As discussed offline, make sure to remove before merging

File: pkg/testdatagen/make_tsp_performance_records.go (hunk 1 above)
Comment (kilbergr) [reply pkg/testdatagen/make_tsp_performance_records.go]: Yep. Do you think this should be explicit

**Comment by kilbergr** [comment]
@breanneboland Hey--you have to recompile first. Otherwise those flags won't work. Did you do that? LMK if so...cause then uh oh.

File: pkg/models/transportation_service_provider_performance.go [hunk 4]
```diff
@@ -17,8 +17,8 @@ import (
 
 var qualityBands = []int{1, 2, 3, 4}
 
-// awardsPerQualityBand struct contains the number of shipments awarded to each tsp according to quality band
-var awardsPerQualityBand = map[int]int{
+// AwardsPerQualityBand is a map of the number of shipments to be awarded per round to each quality band
+var AwardsPerQualityBand = map[int]int{
```
Comment (akostibas) [inline pkg/models/transportation_service_provider_performance.go]: The case of this variable has changed (erroneously)

**Comment by akostibas** [comment]
Also - make sure that the Award Queue can successfully run with the data created by the fake data generator. We should be able to 1) run the fake data generator, 2) run the award queue and see it work.

File: cmd/generate_test_data/main.go [hunk 5]
```diff
@@ -8,11 +8,17 @@ import (
...
+
 // Hey, refactoring self: you can pull the UUIDs from the objects rather than
 // querying the db for them again.
 func main() {
 	config := flag.String("config-dir", "config", "The location of server config files")
 	env := flag.String("env", "development", "The environment to run in, configures the database, presently.")
+	rounds := flag.String("rounds", "none", "Choose none (no awards), full (1 full round of awards), or half (partial round of awards)")
+	numTSP := flag.Int("numTSP", 15, "The number of TSPs you'd like to create")
+	scenarioOne := flag.Bool("scenarioOne", false, "To run test data generation scenario 1.")
+	scenarioTwo := flag.Bool("scenarioTwo", false, "To run test data generation scenario 2.")
```
Comment (akostibas) [inline cmd/generate_test_data/main.go]: Passing "true" as a parameter into a flag is a little nonstandard. Having to write "true" is fairly redundant. Usually a boolean flags are defaulted to false, the fact that you include it in the CLI implies you want it to be true, like: `grep --recursive`. It's not clear to me if the Flag library works like that, you'd have to test.

Another approach would be to rename the flag to `-scenario=` and people supply the name or ID of the scenario they want to use. E.g. `testdatagen -scenario=one`, `testdatagen -scenario=tenshipments`, etc.

File: pkg/testdatagen/make_blackout_date_records.go [hunk 6]
```diff
@@ -53,10 +53,10 @@ func MakeBlackoutDateData(db *pop.Connection) {
 		fmt.Println("TDL ID import failed.")
 	}
 
-	conus := "CONUS_CONUS"
-	market := "market"
 	cos := "BKAS"
-	channel := "dHHG"
+	channel := "CONUS_CONUS"
+	gbloc := "dHHG"
```
Comment (akostibas) [inline pkg/testdatagen/make_blackout_date_records.go]: A GBLOC is a 4 letter code used to identify the JPPSO/PPSO, like `BKAS`.
COS, which is currently `BKAS`, is actually supposed to be a single character, like `D` or `2`.
`dHHG` _might_ be the market, I'd need to double check.

File: pkg/testdatagen/make_tsp_performance_records.go [hunk 7]
```diff
@@ -32,7 +32,7 @@ func MakeTSPPerformance(db *pop.Connection, tsp models.TransportationServiceProv
 }
 
 // MakeTSPPerformanceData creates three best value score records
-func MakeTSPPerformanceData(db *pop.Connection) {
+func MakeTSPPerformanceData(db *pop.Connection, rounds string) {
```
Comment (akostibas) [inline pkg/testdatagen/make_tsp_performance_records.go]: Include what `rounds` means in the function's comment.

File: pkg/testdatagen/make_blackout_date_records.go (hunk 6 above)
Comment (kilbergr) [reply pkg/testdatagen/make_blackout_date_records.go]: I think they were already being misused in that case. I'll change.

File: cmd/generate_test_data/main.go [hunk 8]
```diff
@@ -13,6 +13,10 @@ import (
 func main() {
 	config := flag.String("config-dir", "config", "The location of server config files")
 	env := flag.String("env", "development", "The environment to run in, configures the database, presently.")
+	rounds := flag.String("rounds", "none", "Choose none (no awards), full (1 full round of awards), or half (partial round of awards)")
+	numTSP := flag.Int("numTSP", 15, "The number of TSPs you'd like to create")
```
Comment (akostibas) [inline cmd/generate_test_data/main.go]: This is a little misleading, as it only applies if you're not using a scenario. One option is to update the help text, another is to add a `custom` scenario type.

Comment (kilbergr) [reply cmd/generate_test_data/main.go]: You don't think the readme is enough?

Comment (akostibas) [reply cmd/generate_test_data/main.go]: Not really, because the README is really far away from this script.

**Pull request merged**